NetKet/Utils/json_helper.hpp

```diff
@@ -21,40 +21,50 @@
 #include <string>
 #include <vector>
 
+#include "exceptions.hpp"
+
 namespace netket {
 
 using json = nlohmann::json;
 
-template <class T>
-bool FieldExists(const T &pars, std::string field) {
+template<class Json>
+bool FieldExists(const Json &pars, const std::string& field)
+{
   return pars.count(field) > 0;
 }
 
-template <class T>
-T FieldVal(const T &pars, std::string field) {
-  if (!FieldExists(pars, field)) {
-    std::cerr << "Field " << field << " is not defined in the input"
-              << std::endl;
-    std::abort();
+template<class Json>
+void CheckFieldExists(const Json& pars, const std::string& field, const std::string& context = "")
```

Comment:
can you please add a small comment on the role of "context" here in this function? 
So that we remember in the future what's this about